docs/telemetry.md

```diff
@@ -0,0 +1,50 @@
+Telemetry
+========================
+
+The gaol of the telemetry functionality is collecting and pushing custom business events to the remote analytics server for further processing.
```

Comment:
Can we provide more user-friendly terminology here?
Something like: `Telemetry is a feature that helps us to remotely collect user experience data to make our product better.` And then we can explain that event is a piece of data and continue with existing information.